subiquity/models/filesystem.py

```diff
@@ -2318,8 +2315,9 @@ def can_install(self) -> bool:
     def should_add_swapfile(self):
         mount = self._mount_for_path("/")
         if mount is not None:
-            if not can_use_swapfile("/", mount.fstype):
-                return False
+            if mount.type != "zfs":
```

Comment:
Sorry, I really don't want this bit here.  I want this as part of the ZFS class.
So to abstract that we can add something to Mount and ZFS that isn't a property that allows retrieving the fstype - like the property originally, but with a different name (and not a property) so serialization isn't broken, and then ZFS can handle the special case.